### PR TITLE
Update bazel to 0.29.1, tensorflow to 2.1.0.

### PR DESCRIPTION
**Pull request by riga**
This PR contains the updates to bring tensorflow to version 2.1.0 which allows to pass custom thread pool options to `Session.run()` calls. This is required to finish https://github.com/cms-sw/cmssw/pull/28711.

I added the WIP label as the tensorflow source repository still needs to be updated in tensorflow-sources.spec (see inline comment). I prepared all changes in a personal  branch https://github.com/riga/tensorflow/tree/cms/v2.1.0, which needs to be ported to cms-externals.

### CODE REVIEW

File: tensorflow-sources.file [hunk 1]
```diff
@@ -12,9 +12,10 @@
 BuildRequires: bazel swig java-env git
 Requires: eigen protobuf python py2-numpy py2-enum34 py2-mock py2-wheel py2-keras-applications py2-keras-preprocessing py2-setuptools py2-future
 
-%define tag         c0067ec0da9bcdbc258e447bb4477412ea5e4136
+#%define tag         1a9d6b589044172410202ebca2feb44e5efa1734
```
Comment (riga) [inline tensorflow-sources.file]: Needs to be changed to cms-externals once ported from https://github.com/riga/tensorflow/tree/cms/v2.1.0.

Comment (mrodozov) [reply tensorflow-sources.file]: I'll take care of the "formalities" once the test have passed.

**Comment by mrodozov** [comment]
test parameters
- pull_requests = https://github.com/cms-sw/cmssw/pull/28711

**Review by smuzaffar — APPROVED** [review]

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for slc7_ppc64le_gcc820
just to see if we can build it on ppc64

**Comment by smuzaffar** [comment]
@mrodozov , `bazel` failed to build for ppc64le, can you please do a manual build for aarch64 and see why it failed?

**Comment by mrodozov** [comment]
please test for slc7_aarch64_gcc820
meanwhile, lets try also arm
im checking ppc

**Comment by mrodozov** [comment]
please test for slc7_ppc64le_gcc820
tested it on ppc hope it works on the other archs. lets see
the config patch is fixed in the code: https://github.com/bazelbuild/rules_cc/pull/48/files

**Comment by smuzaffar** [comment]
@mrodozov , looks like tensorflow does not like bazel version 2.0, please move it back to 0.29

**Comment by mrodozov** [comment]
please test for slc7_ppc64le_gcc820

**Comment by smuzaffar** [comment]
closing in favor of https://github.com/cms-sw/cmsdist/pull/5525

**Pull request closed without merging**